### DISCUSSION TITLE
已添加 mihomo 到 homebrew

### DISCUSSION THREAD

**Discussion by guodong000** [MetaCubeX/mihomo#1279] [Show and tell]
https://formulae.brew.sh/formula/mihomo#default

直接通过 brew 安装，使用 brew services 后台启动。

由于 TUN 模式需要 root 权限，可以通过 sudo 运行，或配合 Proxifier 使用。

**Comment by A-caibird** [comment]
我是用 brew 安装使用 mihomo， 代理配置文件在～/.config/mihomo/config.yaml，
但是为什么brew services start mihomo，并没有实现网络代理呢

**Reply by guodong000** [reply]
如果是TUN模式，则需要 root 身份，使用 `sudo brew services start mihomo` 就行了。
然后查看以 root 身份运行的 service，使用 `sudo brew services list`。
停止服务的话就是 `sudo brew services stop mihomo`。
具体可以参考 brew services -h 的帮助文档

> If sudo is passed, operate on /Library/LaunchDaemons or
/usr/lib/systemd/system  (started at boot). Otherwise, operate on
~/Library/LaunchAgents or ~/.config/systemd/user (started at login).

**Reply by A-caibird** [reply]
好的非常感谢， 
还有一个问题就是，我拉去 alpha 分支，build 得到可执行文件， 使用命令行运行
./mihomo -f ~/.config/mihomo/config.yaml --ext-ui ~/.config/mihomo/ui

在 ui 界面查看节点测速都是正常的，但是为什么不能实现系统代理呢

**Reply by guodong000** [reply]
如果TUN模式在macOS下不起作用，一般有两个点：
1. 配置文件问题
2. 如果系统 DNS 为局域网IP，则 mihomo 无法自动劫持 DNS，需要将系统 DNS 改为类似 223.5.5.5 这种公网 IP。具体参考 [mihomo 文档](https://wiki.metacubex.one/config/inbound/tun/#dns-hijack)

**Reply by guodong000** [reply]
另外你可以尝试手动设置系统代理，看看是否可以正常使用

**Comment by A-caibird** [comment]
拉去 alpha 分支代码，go build 后，得到二进制文件
运行./mihomo -f ~/.config/mihomo/config.yaml. --ext-ui ~/.config/mihomo/ui

节点测速正常，但是并未实现网络代理， 打开 Tun 也一样，浏览器访问 google 失败

From: Dong Guo ***@***.***>
Date: Saturday, February 8, 2025 at 5:41 PM
To: MetaCubeX/mihomo ***@***.***>
Cc: Acaibrid ***@***.***>, Comment ***@***.***>
Subject: Re: [MetaCubeX/mihomo] 已添加 mihomo 到 homebrew (Discussion #1279)


如果是TUN模式，则需要 root 身份，使用 sudo brew services start mihomo 就行了。
然后查看以 root 身份运行的 service，使用 sudo brew services list。
停止服务的话就是 sudo brew services stop mihomo。
具体可以参考 brew services -h 的帮助文档

If sudo is passed, operate on /Library/LaunchDaemons or
/usr/lib/systemd/system (started at boot). Otherwise, operate on
~/Library/LaunchAgents or ~/.config/systemd/user (started at login).
